### PR TITLE
More information about departments

### PR DESCRIPTION
**Pull request by jenssegers**
Related Implementation PR: https://github.com/teamleadercrm/core/pull/6753
Related E2E PR: https://github.com/teamleadercrm/api-tests/pull/410

Integrators might want to get information about the company of the user to set up an initial account on their end, or for invoicing reasons.

### CODE REVIEW

**Review by WouterSioen — DISMISSED** [review]

**Review by dietrichm — DISMISSED** [review]

**Review by JordyDeGroeve — DISMISSED** [review]

**Comment by marijnvandevoorde** [comment]
@steftrenson, WDYT: should we add currency here as well while we're at it?

Other than that, looking fine!

**Comment by steftrenson** [comment]
@mrnnn You could already do that, although I have no idea on what the timeframe is to actually have a department in a different currency

**Review by steftrenson — DISMISSED** [review]

**Review by marijnvandevoorde — DISMISSED** [review]
Let's implement this.

So far we have id, name & currency

**Comment by thomasnoe07** [comment]
@jenssegers Can we move this further or it should stay open?

**Comment by luisaterrabrito** [comment]
Why the same information in the list and info?

**Comment by dcsg** [comment]
I agree with @luisabritotl, why do we have the same information in `list` and `info`? Also, why the `website`?

File: apiary.apib [hunk 1]
```diff
@@ -291,7 +291,25 @@ Get a list of departments.
...
+                + address (Address)
+                + emails (array)
+                    + (object)
+                        + type: `primary` (enum[string])
+                            + Members
+                                + primary
+                                + invoicing
+                        + email: `info@piedpiper.eu` (string)
+                + telephones (array[object])
+                    + type: `phone` (enum[string], required)
```
Comment (paultimoce) [inline apiary.apib]: This should be mobile i think since the corresponding database field is gsm

**Review by paultimoce — CHANGES_REQUESTED** [review]

File: apiary.apib (hunk 1 above)
Comment (WouterSioen) [reply apiary.apib]: only mobile and fax for departments for now

Comment (jenssegers) [reply apiary.apib]: In the UI it's called telephone and fax, hence the phone here. Phone also makes more sense on company level (imo).

Comment (WouterSioen) [reply apiary.apib]: 😞

**Comment by jenssegers** [comment]
I see no problem in having the same information returned in both the list and info endpoints. It's basically the same information we're returning here as for the contact/company endpoints.

**Comment by dcsg** [comment]
@jenssegers why iban, website, bic and that type of information is available in a `list` endpoint? It belongs to the **details** of the `resource` hence in the `info` endpoint. Otherwise, why do we have two endpoints? We could just have the `list` one.

File: apiary.apib [hunk 2]
```diff
@@ -308,7 +326,25 @@ Get details for a single department.
...
+                    + email: `info@piedpiper.eu` (string)
+            + telephones (array[object])
+                + type: `phone` (enum[string], required)
+                    + Members
+                        + phone
+                        + fax
+                + number: `092980615` (string, required)
+            + website: `https://piedpiper.com` (string)
             + currency: `EUR` (string)
+            + iban: `BE12123412341234` (string)
```
Comment (dcsg) [inline apiary.apib:329]: B/Q: why are we adding this to the root level? This for me seems to be missing a "bank information" object and then having these fields inside that object.

Comment (dietrichm) [reply apiary.apib:329]: Lots of endpoints are adding IBAN and BIC on the top level right now. This might be a BC breaking improvement that we can release in a next API version.

**Comment by luisaterrabrito** [comment]
I don't agree, @jenssegers. I understand more information might be added in the future to the info endpoint, but for now, I find it confusing, I wouldn't know which one to use. Feels like there is no need to have both.

**Comment by jenssegers** [comment]
Ok let's reduce the information on the list endpoint then. Easier to add it later than to remove it afterwards. Any other suggestions for fields to drop other than the iban/bic?

**Comment by luisaterrabrito** [comment]
I don't know where it is/will be used, but making a comparison with the invoices list that is used in the redesign to show all the invoices, we only need some of the basics, I would remove the addresses, emails, telephones, website or just leave the default ones and show the others in the info. The emails have one that is primary, I guess we could return that one and if the department has more of them show it in the info

**Comment by luisaterrabrito** [comment]
Please leave the currency, I think @lowiebenoot might be using it

**Comment by lowiebenoot** [comment]
I'm indeed using the currencies on `departments.list`.

**Comment by WouterSioen** [comment]
Who's following up to make the specs correct? @jenssegers ?

**Review by WouterSioen — DISMISSED** [review]
I cleaned up the .list endpoint and got rid of a lot of fields.

Is this good to go now?

**Review by steftrenson — DISMISSED** [review]

**Review by JordyDeGroeve — DISMISSED** [review]

**Review by dietrichm — DISMISSED** [review]

File: apiary.apib [hunk 3]
```diff
@@ -291,6 +291,8 @@ Get a list of departments.
             + (object)
                 + id: `67c576e7-7e6f-465d-b6ab-a864f6e5e95b` (string)
                 + name: `Human Resources` (string)
+                + vat_number: `BE0899623035` (string)
+                + website: `https://piedpiper.com` (string)
```
Comment (dcsg) [inline apiary.apib]: Q: why is this necessary in the `list`? My concern is that this field is not required and I would imagine that most of the times it will be `null`?

Comment (WouterSioen) [reply apiary.apib]: I can imagine that you'd like to show that in an overview of departments?

Comment (WouterSioen) [reply apiary.apib]: Okay, you're right, website is filled in for only 15% of departments and is not really prominent in the UI, so it'll be `null` a lot of the times.

Not sure if this is an issue though? I can still imagine this getting a more prominent place later?

Comment (WouterSioen) [reply apiary.apib]: We could still add the fields at that moment though.

Comment (dcsg) [reply apiary.apib]: The thing is if it's only 15% than the UI will have an empty column 85% of the times will be awkward for the user IMO other than that having the field would be ok ;)

**Review by dcsg — APPROVED** [review]

**Review by luisaterrabrito — APPROVED** [review]

**Review by paultimoce — APPROVED** [review]

**Comment by WouterSioen** [comment]
Let's consider this proposal final so we can start on the implementation.

**Comment by luisaterrabrito** [comment]
Sorry guys, I thought I was on my PR
